src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -645,7 +678,7 @@ bool DomainParticipantImpl::register_type(
 
     if (type->auto_fill_type_object() || type->auto_fill_type_information())
     {
-        register_dynamic_type_to_factories(type);
+        //register_dynamic_type_to_factories(type);
```

Comment:
I thought this PR was just about this. Is this creating a leak?

Reply:
oh, that was changed during the debugging and I forgot it. That instruction should be activated, I'll change that.